src/com/facebook/buck/json/buck.py

```diff
@@ -201,9 +204,39 @@ def add_rule(rule, build_env):
             # the user to re-run `buck autodeps` in this scenario. Unfortunately, we do not have
             # a mechanism to relay warnings from buck.py at the time of this writing.
             pass
+    if rule['buck.type'] in ('apple_library', 'apple_binary'):
+        filter_and_generate_swift_rule(rule, build_env)
     build_env.rules[rule_name] = rule
 
 
+def filter_and_generate_swift_rule(rule, build_env):
```

Comment:
I think this shouldn't change the parser: if you wish to make changes to how the graph is interpreted they should be done in the target graph to action graph transformation phase, not in the parser.

An alternative would be to make the transformation in the repo's build files through common `include_defs` helpers.